metaseq/dataclass/configs.py

```diff
@@ -737,3 +748,49 @@ class MetaseqConfig(MetaseqDataclass):
     lr_scheduler: Any = MISSING
     bpe: Any = MISSING
     tokenizer: Any = None
+
+
+class DynamicConfig(dict):
+    """
+    can be used instead of redis to store updatable key-value
+
+    test:
+        data = {"a": 1}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        dcfg = DynoCfg(json_file_path = "data.json", timeout = 1)
+        print(dcfg["a"]) # > 1
+        data = {"a": 4}
+        with open('data.json', 'w') as fp:
+            json.dump(data, fp)
+        print(dcfg["a"]) # > 1
+        time.sleep(1)
+        print(dcfg["a"]) # > 4
+    """
+
+    default_state = {"force_profile": False}
+
+    def __init__(self, json_file_path=None, timeout=30):  # timeout in sec
+        self.timeout = timeout
+        self.timer_start = time.time()
+        self.json_file_path = json_file_path
+        if self.json_file_path is not None:
+            with open(self.json_file_path, "r") as json_file:
+                super().__init__(**json.load(json_file))
+        else:
+            super().__init__(**DynamicConfig.default_state)
+
+    def refresh(self):
```

Comment:
Is there any situation in which we'd need this to be thread safe?

Reply:
What do you mean thread safe? Like if this code is executed in multiple thread at once? that is not intended but I also don't see how this would break if it does

Reply:
> Like if this code is executed in multiple thread at once? 

Exactly that. The issue would be that `self.timer_start` isn't managed atomically. It seems like the worst case would be that the file is read multiple times, so it's probably fine to ignore.

Reply:
It is intended that the files are read as many times as there are processes or threads - this way every worker gets updated dictionary. I still don't see problems here